src/main/java/Duke.java

```diff
@@ -1,10 +1,250 @@
+        } catch (InvalidIndexException e) {
+            System.out.println("\t☹ OOPS!!! I can't find this task");
+        } catch (EmptyNameException e) {
+            System.out.println("\t☹ OOPS!!! The description of a todo cannot be empty.");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (finishSetup) {
+            System.out.println(DOTTED_LINE);
+        }
+    }
+
+    private static void executeCommand(String input, String command, boolean finishSetup)
+            throws InvalidIndexException, EmptyNameException, IOException {
+        switch (command) {
+        case LIST_COMMAND: {
+            printList();
+            break;
+        }
+        case DONE_COMMAND: {
+            int index = getIndexFromInput(input, command);
+            markTaskAsDone(index, finishSetup);
+            if (finishSetup) {
+                addInputToFile(input);
+            }
+            break;
+        }
+        case DELETE_COMMAND: {
+            int index = getIndexFromInput(input, command);
+            removeTask(index, finishSetup);
+            if (finishSetup) {
+                addInputToFile(input);
+            }
+            break;
+        }
+        case TODO_COMMAND: {
+            String name = input.substring(TODO_COMMAND.length()).trim();
+            if(name.length() == 0) {
+                throw new EmptyNameException();
+            }
+            Task todo = new Todo(name);
+            addToTaskList(todo, finishSetup);
+            if (finishSetup) {
+                addInputToFile(input);
+            }
+            break;
+        }
+        case DEADLINE_COMMAND: {
+            int byIndex = input.indexOf(BY_COMMAND);
+            if (byIndex == -1) {
+                System.out.println("Please tell me when this is due!");
+                return;
+            }
+            String name = input.substring(DEADLINE_COMMAND.length(), byIndex).trim();
+            String by = input.substring(byIndex + BY_COMMAND.length()).trim();
+            Task deadline = new Deadline(name, by);
+            addToTaskList(deadline, finishSetup);
+            if (finishSetup) {
+                addInputToFile(input);
+            }
+            break;
+        }
+        case EVENT_COMMAND: {
+            int atIndex = input.indexOf(AT_COMMAND);
+            if (atIndex == -1) {
+                System.out.println("Please tell me when is this event!");
+                return;
+            }
+            String name = input.substring(EVENT_COMMAND.length(), atIndex).trim();
+            String at = input.substring(atIndex + AT_COMMAND.length()).trim();
+            Task event = new Event(name, at);
+            addToTaskList(event, finishSetup);
+            if (finishSetup) {
+                addInputToFile(input);
+            }
+            break;
+        }
+        default: {
+            System.out.println("Something went wrong here...");
+            break;
+        }
```

Comment:
Good job following the coding standard to include the breaks, but a further improvement could be to follow the guide by removing the braces encapsulating each case

Reply:
If I remove the braces, I get this error:
Variable 'index' is already defined in the scope

Or should I include the braces only when it is needed? (E.g. the default case doesn't need to have braces)